### ISSUE TITLE
Wildcard matching in OSC path configuration?

### ISSUE THREAD

**Issue by dmeltzer** [cbdevnet/midimonster#11]
Hello,

I found myself today needing to output a lot of similar osc paths.

Basically:
```
/channel/1/red = f 0.0 1.0
/channel/1/green = f 0.0 1.0
/channel/1/blue = f 0.0 1.0
/channel/2/red = f 0.0 1.0
/channel/2/green = f 0.0 1.0
etc
```

It would be great if there was a way of doing something like
```
/channel/*/red = f 0.0 1.0

or even better
/channel/*/[red,green,blue] = f 0.0 1.0
```

But I'm not sure of a good way to do that that does not interfere with those characters being part of the path--or even if that format is the right way to do things.  I'd like to look at implementing something like this, but wanted to get some thoughts first.

**Comment by cbdevnet** [comment]
Hi, and thanks for the great work!

There's actually a comment in the code for that kind of thing at https://github.com/cbdevnet/midimonster/blob/master/osc.c#L722 , though it's really just a FIXME.

But yes, with proper OSC path matching, that would be possible.

Currently, path preconfiguration in the instance configuration creates an entry in the instance's `channels` array (this is at https://github.com/cbdevnet/midimonster/blob/master/osc.c#L426 and should probably be factored out into a procedure by itself, because as it is it looks pretty bad). Currently, the index of a channels configuration within that array is used as instance-global channel identifier (with some magic dust for multi-parametric channels).

This could (and probably should) be extended to allow just the behaviour you described.
One particular thing to watch for are the aforementioned multi-parametric channels though, the implementation is a bit brittle as it is and could probably use some decoupling - though then again, holding too much state is not too good either (Holding an entire array of channel-pointers is probably not the best solution I think)

As for using `*` within the path, the [OSC spec](http://opensoundcontrol.org/spec-1_0) actually says exactly what you describe in the "OSC Message Dispatching and Pattern Matching" section (and that why the comment is there in the code ;))

**Comment by cbdevnet** [comment]
Implemented in 26ee2eacc7d60aa379c9e4b9b9c6b8bcdcd4bc6b.

See https://github.com/cbdevnet/midimonster/blob/master/configs/flying-faders.cfg for an example ;)
